watsononlinestore/watson_online_store.py

```diff
@@ -533,9 +632,9 @@ def slack_encode(input_text):
 
             product_data = {
                 "cart_number": str(cart_number),
-                "name": slack_encode(get_product_name(result, data_source)),
-                "url": slack_encode(get_product_url(result, data_source)),
-                "image": slack_encode(get_image_url(result, data_source)),
+                "name": slack_encode(get_product_name(result)),
+                "url": slack_encode(get_product_url(result)),
+                "image": slack_encode(get_image_url(result)),
```

Comment:
so maybe someday these could be like slack_encode(data_source.get_xxx(result))
and we'd have less of the if/else stuff. But I'd merge this first and do that separately if anyone gets to it.